rfcs/text/0000-data-citation-plan.md

```diff
@@ -0,0 +1,151 @@
+  - Upload the matrix output file(s) for this version of the project to the DOI repository.
+  - Update a project description to the DOI repository (e.g. what’s in this version?).
+  - Create a link in the DOI repository back to the project details page in the Data Browser.
+
+### Acceptance Criteria [optional]
+
+#### Phase 1
+  - A citation link is provided in the Data Browser for each project
+  - The citation link remains valid for the lifetime of the DCP
+  - The citation link is a URL which resolves to the latest version of the project details page for the specified project
+  - The citation URL is used in publications by scientists to cite DCP project data
+#### Phase 2 (in addition to Phase 1 criteria)
+  - The Data Operations team is able to create an immutable citation reference for each project in a Data Release
+  - The citation reference consists of a versioned DOI
+  - The Data Operations team can update the version of the project's DOI in a later Data Release
+  - The Data Browser provides a means of downloading the data associated with a specific version of a project from a Data Release
+#### Phase 3 (in addition to Phase 1 and 2 criteria)
+  - An authorized user may create a citation for an arbitrary set of versioned data selected via the Data Browser
+  - An authorized user may update a previously versioned set of data to a new version and update the version of the corresponding citation
+   - An authorized user may share a citation which they have created with others users who can then access the specific version of the data referenced by the citation
+
+### Unresolved Questions
+
+Must a data citation that provides a DOI provide an *immutable* view of all the data and metadata associated with a project? Or is acceptable for any of the following to undergo additions, updates or deletions over time?
+  - a project's primary data
+  - a project's secondary analysis outputs
+  - a project's expression matrix outputs, generated by the Matrix Service
+  - a project's metadata
+  
+For Phase 2 & 3 does the Data Browser need to provide a view of the cited data on which further faceted searches can be performed?
+Currently the Data Browser provides only a view of the latest data and metadata. It may be desirable to add another "facet" such as "Release Version" which would enable the user to view a particular point-in-time Data Release and then perform further faceted searches within that view.
+
+Can a DOI for the DCP resolve to an external website? The author(s) will work with the UX team to come up with pros and cons of using an external authority vs setting up the DCP as a DOI assigning entity and then, if that effort doesn't give us a clear answer, we will ask the Oversight Committee for their view.
+
+### Drawbacks and Limitations [optional]
+
+*Why should this RFC **not** be implemented?*
+
+### Prior Art [optional]
+
```

Comment:
Some emphasis on the use cases of programatic consumers, who will likely be your highest frequency users, would be good. Through the lens of current best practice you have only indirectly (through BioStudies) mentioned compact identifiers, but these are critical and would help at all phases of delivery. Consideration of best practice should go into the prior art section.

DOI and URL have both been mentioned. These are relevant to publishers and a user with a keyboard. But compact identifiers are most frequently used to consume, map, link and share data and to my mind represent best practice. They are also the most useful way to discuss a project via word of mouth or via slack which is essential for developers, wranglers and data ops. Saying 'prof x's latest organ y dataset' is not scalable and uuids are not human readable or memorable, which increases the likelihood of miscommunication (missing use case). Within the EBI, these identifiers are used to make data interoperable but all major biological data institutions also embrace them. As a specific example the successful INSDC (sharing genomic data between NCBI, EBI and DDBJ) works by mapping these identifiers.

You will have come across identifiers that look like this in our metadata SRAxxxx, GEOxxxx, PMCxxxx, PRJxxxx... We use them to reference sources of data. This is the same use case as your downstream users. This single short identifier can be resolved similar to a DOI through identifiers.org, communicated in metadata easily and thanks to the resource specific prefix bakes in the source of the data. Bioinformaticians are used to dealing with these identifiers in this format.

As it stands uuids are not HCA specific. Thankfully no-one else (in our realm) seems to use them so we (at SCEA) guess these identifiers refer to a HCA dataset. It would be preferable to have an ID HCAxxxx to confirm this.

At the moment our metadata would look like this:
`"data source" : ["ERP114453", "PRJEB31843", "c4077b3c-5c98-4d26-a614-246d12c2e5d7"]`
Spot the odd one out? How do I resolve this if anyone other than the HCA starts to use uuids?

Other resources also support these identifiers for different granularities of the data. For example ENA provide prefixes to describe studies, experiments, samples and runs. This is very useful for downstream users like us who want to split and splice datasets but still need to refer to the provenance of the data.

So compact identifiers give you a familiar way to communicate granularity and source of data while easing programatic or manual access. These identifiers are often used in the literature to refer to data rather than DOIs. If the plan is for BioStudies to mint these identifiers for HCA projects, I would ask the following:

- This functionality is added ASAP. Once a referencing scheme is established it is difficult and costly to change. If uuids/URLs are referenced in upcoming papers the trend will quickly stick.
- This identifier is in the metadata before a dataset is released so that it can always be relied upon downstream.
- The prefix maintains provenance so it is clear the dataset is HCA
- HCA is able to provide references to more granular components of a project (sample, assay etc) in the future.

Reply:
The project metadata does contain compact identifiers in the form of "project.insdc_project_accessions" (e.g. SRP078321), "project.insdc_study_accessions" (e.g. PRJNA328774), "project.array_express_accessions" (e.g. E-GEOD-84133), "project.geo_series_accessions" (e.g. GSE84133), and "project.biostudies_accessions".
@lauraclarke Are you on board with adding HCA-specific accessions (e.g. HCAxxxxx) to the project metadata and possibly to other objects (e.g. samples, assays)? Is biostudies_accessions ever populated (I couldn't find an example)? Noting that the INSDC/GEO/ArrayExpress accessions are not always populated for a project and don't provide the direct reference to data in the DCP that would be provided by an HCA compact identifier.

@hewgreen for programmatic consumers, are you expecting additions to an existing API or proposing a new service, and what functions would be needed?

Reply:
I hate to cite a paper I am myself an author on, but it is worth reading this article before deciding whether or not to create new accessions: https://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.2001414. This article also references many other useful papers on identifying resources in the life sciences.

Reply:
@tburdett beat me to it but to agree with him. I think we need to think through the consequences of adding another namespace to the identifiers within bioinformatics. There are upsides to do it but there are also costs and we need to ensure we understand them before we embark down that route.

Reply:
@theathorn As you rightfully point out the compact identifiers you refer to are to secondary resources. They do not reflect HCA data (which can be novel, analysed or truncated). They are not always present and cannot be relied upon. Therefore, these do not address the requirements I laid out.

@tburdett the paper outlines the considerations perfectly. One interesting idea is human readable labels. Maybe these could solve some issues (not user provided titles). But overall I think it suggests we should accession in some cases and think carefully about identifier design. UUIDs maybe not being great. I recommend the paper to others. My interpretation of the 'lessons are as follows:

**1. Credit any derived content using its original identifier**
We should create IDs for HCA's new data and new analysis because it's novel stuff. Source IDs are important.

**2. Help local IDs travel well: Document prefix and patterns**
Prefixes are super important. aka a uuid alone isn't useful.

**3. Opt for simple, durable web resolution**
We should get a resolution provider. Why I mention identifiers.org but DOI sort of works for this although you have to follow them to work out what they are. Our URL prefixes are already nice and simple but the local ID's are long, cumbersome, human unrepeatable.

**4. Avoid embedding meaning or relying on it for uniqueness:**
We don't want to pack meaning into our IDs. Fair point. I don't think anyone want's this but we should be careful not to.

**5. Design new identifiers for diverse uses by others**
The ID's need to be designed to be useful. UUIDs cannot be described as user friendly. We could design something much nicer.

**8. Make URIs clear and findable & 10. Reference and display responsibly**
Our prefixes aren't in the metadata as passed from the DSS. Should they be if they are the only way downstream consumers know a uuid belongs to the HCA?

The following lessons I think we have learned and are working towards already:
**6. Implement a version-management policy**
**7. Do not reassign or delete identifiers**
**9. Document the identifiers you issue and use**

Reply:
I have proposed a F2F session on Accessions, fyi, with a proposal to put them in use on HCA data.